department-of-veterans-affairs/va.gov-cms
# Add VA Form content type.

## Description
Adds a new content type, `VA Form` and supporting views, including an admin screen (http://pr1774.ci.cms.va.gov/admin/content/va-forms).

## Testing done
 - Field requirements reviewed
 - Create & edit screen UX review completed.
 - Behavior controlling editability of migrated fields tested. If one of the following fields is empty, they all become editable:
    - title
    - field_va_form_administration
    - field_va_form_number
    - field_va_form_title
    - field_va_form_issue_date
    - field_va_form_url

## QA Steps
1. Create a VA Form node by going to http://pr1774.ci.cms.va.gov/node/add/va_form and adding values for 'Form Name' and fields within 'Editable forms data'.
    - Confirm that the node is created.
    - Confirm that all the fields required for forms search and a minimal VA Form page are present.
1. Edit the node.
    - Confirm that migrated fields are not editable but their values are displayed under 'Forms data'.
    - Confirm that non-migrated fields can be entered appropriately.
1. Go to http://pr1774.ci.cms.va.gov/admin/content/va-forms and confirm that the node appears appropriately.

## Screenshots

Create VA Form screen with 'migrated' data
<img width="1193" alt="Screen Shot 2020-06-09 at 3 43 02 PM" src="https://user-images.githubusercontent.com/7660109/84192727-6a99b380-aa68-11ea-86e2-98e88ba42a78.png">

Edit VA Form screen with 'migrated' data
<img width="1532" alt="Screen Shot 2020-06-09 at 3 44 32 PM" src="https://user-images.githubusercontent.com/7660109/84192880-9fa60600-aa68-11ea-8be6-655df7487dc4.png">

VA Form listing page
<img width="1468" alt="Screen Shot 2020-06-09 at 3 58 05 PM" src="https://user-images.githubusercontent.com/7660109/84194174-a59ce680-aa6a-11ea-98d4-dacceab14e0c.png">
